app/controllers/auth_controller.rb

```diff
@@ -0,0 +1,33 @@
+class AuthController < ApplicationController
+  def login
+    user = User.find_by(username: login_params[:username])
+    if user && user.authenticate(login_params[:password])
+      token = JWT.encode({ user_id: user.id }, Rails.application.secret_key_base, 'HS256')
+      render json: { user: user, token: token }
+    else
+      render json: {}, status: 401
+    end
+  end
+
+  def persist
+    if request.headers['Authorization']
+      encoded_token = request.headers['Authorization']
+      begin
+        token = JWT.decode(encoded_token, Rails.application.secret_key_base, true, algorithm: 'HS256')
```

Comment:
I would encapsulate both token encoding and decoding. That way the JWT logic and dependency will be in one place. It would also help reduce the logic in your controller.

Reply:
Good point. This will be encapsulated in a separate pull request.